ovn-tester/cms/openstack/openstack.py

```diff
@@ -0,0 +1,561 @@
+        project.networks.append(neutron_int_network)
+
+        self._add_router_port_internal(
+            neutron_int_network, project.router, project
+        )
+
+        snated_network = DualStackSubnet(int_net.cidr)
+        if snat_port is not None:
+            self.nbctl.nat_add(project.router, snat_port.ip, snated_network)
+
+    def _create_project_net(self, net_name: str, mtu: int = 1500) -> LSwitch:
+        """Create Logical Switch that represents neutron network.
+
+        This method creates Logical Switch with same parameters as Neutron
+        would.
+        """
+        switch_ext_ids = {
+            "neutron:availability_zone_hints": "",
+            "neutron:mtu": str(mtu),
+            "neutron:network_name": net_name,
+            "neutron:revision_number": "1",
+        }
+        switch_config = {
+            "mcast_flood_unregistered": "false",
+            "mcast_snoop": "false",
+            "vlan-passthru": "false",
+        }
+
+        neutron_net_name = f"neutron-{uuid.uuid4()}"
+        neutron_network = self.nbctl.ls_add(
+            neutron_net_name,
+            self.next_int_net(),
+            ext_ids=switch_ext_ids,
+            other_config=switch_config,
+        )
+
+        return neutron_network
+
+    def _add_network_subnet(self, network: NeutronNetwork) -> None:
+        """Create DHCP subnet for a network.
+
+        Adding subnet to a network in Neutron results in DHCP_Options being
+        created in OVN.
+        """
+        external_ids = {
+            "neutron:revision_number": "0",
+            "subnet_id": str(uuid.uuid4()),
+        }
+        static_routes = (
+            f"{{169.254.169.254/32,{network.gateway}, "
+            f"0.0.0.0/0,{network.gateway}}}"
+        )
+        options = {
+            "classless_static_route": static_routes,
+            "dns_server": "{1.1.1.1}",
+            "lease_time": "43200",
+            "mtu": "1442",
+            "router": str(network.gateway),
+            "server_id": str(network.gateway),
+            "server_mac": str(RandMac()),
+        }
+        dhcp_options = self.nbctl.create_dhcp_options(
+            str(network.network.cidr), ext_ids=external_ids
+        )
+
+        self.nbctl.dhcp_options_set_options(dhcp_options.uuid, options)
+
+    def _assign_port_ips(self, network: NeutronNetwork) -> None:
+        """Assign IPs to each LS port associated with the Network."""
+        for uuid_, port in network.ports.items():
+            if port.ip is None or port.ip == "unknown":
+                updated_port = self.nbctl.ls_port_set_ipv4_address(
+                    port, str(network.next_host_ip())
+                )
+                network.ports[uuid_] = updated_port
+                # XXX: Unable to update external_ids
+
+    def _add_network_port(self, network: LSwitch, project_id: str) -> LSPort:
+        """Create regular port in LSwitch with Neutron's external IDs."""
```

Comment:
The settings for the port in this method looks like they come from the special Neutron OVN Metadata port.

For a regular VM port the type would be `""` and the `requested-chassis` would point at an actual node.

Reply:
I think I just misunderstood the port that was created and thus I named it wrong. When user runs `openstack network create`, Neutron automatically creates one `LSPort` for the `LSwitch`. I wasn't entirely clear on the purposed of this port so I gave it this general name/description. Do you have any suggestion on naming/purpose of this port?